core/src/main/java/com/linecorp/armeria/server/DefaultUnhandledExceptionsReporter.java

```diff
@@ -93,28 +93,13 @@ public void serverStopped(Server server) throws Exception {
     }
 
     private void reportException() {
-        final long totalExceptionsCount = counter.sum();
-        final long newExceptionsCount = totalExceptionsCount - lastExceptionsCount;
-        if (newExceptionsCount == 0) {
-            return;
-        }
-
-        final Throwable exception = thrownException;
-        if (exception != null) {
+        for (ExceptionContext exceptionContext : exceptionStats.dump()) {
             logger.warn("Observed {} exception(s) that didn't reach a LoggingService in the last {}. " +
                         "Please consider adding a LoggingService as the outermost decorator to get " +
                         "detailed error logs. One of the thrown exceptions:",
-                        newExceptionsCount,
-                        TextFormatter.elapsed(intervalMillis, TimeUnit.MILLISECONDS), exception);
-            thrownException = null;
-        } else {
-            logger.warn("Observed {} exception(s) that didn't reach a LoggingService in the last {}. " +
-                        "Please consider adding a LoggingService as the outermost decorator to get " +
-                        "detailed error logs.",
-                        newExceptionsCount, TextFormatter.elapsed(intervalMillis, TimeUnit.MILLISECONDS));
+                        exceptionContext.getCounter(),
+                        TextFormatter.elapsed(intervalMillis, TimeUnit.MILLISECONDS),
+                        exceptionContext.getException());
```

Comment:
This logic is now outdated and will not look good enough.

- It will now print "Observed ... Please consider ..." way too many times.
- "One of the thrown exceptions" doesn't make sense because we now record all stack traces.

What do you think about rewriting this part so it generates more comprehensive log?